connectors/src/connectors/microsoft/temporal/workflows.ts

```diff
@@ -131,6 +133,18 @@ export async function microsoftDeletionWorkflow({
   await microsoftDeletionActivity({ connectorId, nodeIdsToDelete });
 }
 
+export async function microsoftGarbageCollectionWorkflow({
+  connectorId,
+}: {
+  connectorId: number;
+}) {
+  let idCursor = 0;
+  while (idCursor !== null) {
+    idCursor = await microsoftNodesGCActivity({ connectorId, idCursor });
+    await sleep("1 minute");
```

Comment:
should theoretically free the worker (workflow is executed on temporal server not on our workers)
checking with you @tdraier since IIRC you mentioned it could actually be an issue?

Reply:
Finally checked where/how the workflow are executed, and they do are in the same worker as our activities (it's using the queue passed in client.workflow.start - we could use another worker/queue). However they are run in a v8 virtual machine context with `vm.runInContext`. If sleep releases the worker it's fine.

Reply:
:+1: https://docs.temporal.io/develop/java/timers seems to mean that ("does not tie up the process")